Add support for fetching multiple pair addresses in a single request (up to 30)

paramkr2 opened the pull request:
Dexscreen API allows fetching up to 30 pair addresses in a single request.

[API Documentation Reference](https://docs.dexscreener.com/api/reference#:~:text=separated%20pair%20addresses-,(up%20to%2030%20addresses),-E.g.%3A%200xAbc1)

* Updated the get_token_pair and get_token_pair_async methods in the DexscreenerClient class to handle both single and multiple pair addresses.

```
multiple_pairs = await client.get_token_pair_async("ethereum", [
        "0xC2aDdA861F89bBB333c90c492cB837741916A225",
        "0x7BeA39867e4169DBe237d55C8242a8f2fcDcc387"
    ])
```

paramkr2 committed:
Get data with multiple pairaddress

nixonjoshua98 commented:
A separate method would be preferred `get_token_pairs_async` and you have a print statement left in. Type hinting it as a generic iterable like tuple, list, etc would be more accurate as well as these would all work

paramkr2 committed:
Seprated get_token_pair_list

paramkr2 commented:
I've made the following updates based on the feedback

* Added a separate method `get_token_pair_list_async` because `get_token_pairs_async` is already used for other functionality.

* Updated type hinting to be more flexible by using `Iterable` instead of `List`.
